### ISSUE TITLE
macOS custom client requests unnecessary privileges 

### ISSUE THREAD

**Issue by JohnnyHenver** [rustdesk/rustdesk#8687]
### Bug Description

A custom macOS client which was configured with option "outgoing only" will still ask for privileges related to incoming connections in a very prominent manner. 

The client states "_You can create connections to other devices, but other devices can not establish a connection to your device_" yet right below in bright colours "_To control your PC remotely, you need to assign permissions to rustdesk_"

Ideally this warning should be removed, as it is confusing to users and my promote them to grant unnecessary privileges.

### How to Reproduce

1. Setup rustdesk server pro
2. Create custom client for macOS (specify "outgoing" under connection type)
4. Install the custom client on macOS
5. Run the client

### Expected Behavior

The client should not ask for privileges which are not required according to its configuration.

### Operating system(s) on local side and remote side

macOS Sonoma 14.5

### RustDesk Version(s) on local side and remote side

1.2.6

### Screenshots

![iScreen Shoter - tt-rustdesk - 240712105303](https://github.com/user-attachments/assets/0c3d5af4-0585-427e-9578-c9c9a1b4e3fe)

![iScreen Shoter - Google Chrome - 240712105455](https://github.com/user-attachments/assets/5fd94d78-9df5-4941-ba5a-584e6ce91a6f)

**Comment by rustdesk** [comment]
https://rustdesk.com/docs/en/self-host/client-configuration/advanced-settings/#hide-username-on-card

**Comment by Loremipsumdolorsitametconsetetur** [comment]
Hi @rustdesk setting "hide-help-cards=Y" on an incoming-only client causes the UAC on Windows 11 to not be forwarded to the controlling side effectively breaking UAC transmission and causing a frozen screen until the user on the controlled side finishes interacting with the UAC. 
Do you need more information or a new issue?

**Comment by rustdesk** [comment]
@Loremipsumdolorsitametconsetetur could you upload a video for this? I could not understand quite much. If you create a new issue, please record a video for it. From the source code, "hide-help-cards=Y" has nothing to do with this forwarding. It is quite abnormal.

**Comment by Loremipsumdolorsitametconsetetur** [comment]
Hi @rustdesk , sorry for not being clear. Let me try again: When executing CMD as admin in an active RustDesk session like this the behaviour is different depending on the advanced setting in custom clients: 

<img width="402" height="415" alt="Image" src="https://github.com/user-attachments/assets/894a39d5-dce9-42eb-8de2-b7ce1c9da727" />

"hide-help-cards=N" --> I see the UAC pop up and can enter username + password
"hide-help-cards=Y" --> no UAC pop up, screen freezes and no UAC is visible (just like with portable RustDesk with no elevation -> https://rustdesk.com/docs/en/client/windows/windows-portable-elevation/)

If you still want a video, let me know.

**Comment by rustdesk** [comment]
@21pages please follow it up

**Comment by 21pages** [comment]
I can't reproduce it with the incoming-only client. @Loremipsumdolorsitametconsetetur , can you double-check that the version with `hide-help-cards=Y` is installed?

**Comment by Loremipsumdolorsitametconsetetur** [comment]
@21pages thank you for your effort. I am currently doing more testing, collect more information about the issue and get back to you in the next few days.

**Comment by Loremipsumdolorsitametconsetetur** [comment]
@21pages thanks again for your patience.

We’ve now tested the scenario thoroughly across multiple devices and can confirm that the issue was not caused by RustDesk.

The root cause was a broken installation on our side: our deployment solution prevented the RustDesk service from starting during installation. Because of this, anything that didn’t rely on the service functioned as if RustDesk were running in portable mode — which is also why UAC elevation didn’t work.

During manual testing, we mistakenly assumed the issue was related to "hide-help-cards=Y". In reality, the only difference was that manual installation correctly started the service, which made the UAC problem disappear. This caused our incorrect report in this issue last week.

In short — the fault was entirely on our side.
